src/averageAges.js

```diff
@@ -36,8 +58,35 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  let result = 0;
+  const avAge = 0;
+  const woman = people.filter(el => (el.sex === 'f'));
+  let womanAvAge = woman.reduce(
+    (accum, curr) => accum + (curr.died - curr.born),
+    avAge);
+
+  womanAvAge /= woman.length;
+
+  function isMother(e) {
+    const motherCheck = people.find(el => e.name === el.mother);
+
+    return motherCheck !== undefined;
+  }
+
+  const mothers = woman.filter(isMother, woman);
+
+  let mothersAvAge = mothers.reduce(
+    (accum, curr) => accum + (curr.died - curr.born),
+    avAge);
+
+  mothersAvAge /= mothers.length;
+
+  (withChildren === undefined) ? result = womanAvAge : result = mothersAvAge;
```

Comment:
Same here, you should check the second argument at the start of the function
And create correct array of people depending on it